### PR TITLE
Api key auth strategy

### PR DESCRIPTION
**Pull request by AleF83**

### CODE REVIEW

File: services/src/modules/authentication/strategies/api-key.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const headerName = config.header;
+  if (headerName) {
+    apiKey = request.headers[headerName] as string;
+  }
+  if (!apiKey) {
+    const queryParamName = config.queryParam;
+    if (queryParamName) {
+      apiKey = request.query[queryParamName] as string;
+    }
+  }
```
Comment (tomeresk) [inline services/src/modules/authentication/strategies/api-key.ts:20]:
```suggestion
  const apiKey = (request.headers[config.header] || request.query[config.queryParam]) as string
```

File: services/src/modules/authentication/strategies/api-key.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  if (!apiKey) {
+    throw new Error('Unauthorized');
+  }
```
Comment (tomeresk) [inline services/src/modules/authentication/strategies/api-key.ts:23]:
```suggestion
  if (!apiKey) throw new Error('Unauthorized');
```

File: services/src/modules/authentication/strategies/api-key.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  reqLogger.trace('Api key found in request');
+
+  const apiKeyConfig = apiKeys[apiKey];
+  if (!apiKeyConfig) {
+    reqLogger.debug('Unexpected api key');
+    throw new Error('Unauthorized');
+  }
+  reqLogger.trace({ name: apiKeyConfig.name }, 'Api key found in configuration');
+
+  if (apiKeyConfig.enabled === false) {
```
Comment (tomeresk) [inline services/src/modules/authentication/strategies/api-key.ts:34]:
```suggestion
  if (!apiKeyConfig.enabled) {
```

File: services/src/modules/authentication/strategies/api-key.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    reqLogger.debug('The api key is disabled');
+    throw new Error('Unauthorized');
+  }
+
+  const anonymousPaths = config.authenticatedPaths;
```
Comment (tomeresk) [inline services/src/modules/authentication/strategies/api-key.ts:39]: I don't understand why `anonymousPaths` comes from the `authenticatedPaths` config value

**Review by tomeresk — CHANGES_REQUESTED** [review]

File: services/src/modules/authentication/strategies/api-key.ts (hunk 3 above)
Comment (AleF83) [reply services/src/modules/authentication/strategies/api-key.ts:34]: Replaced `enabled` with `disabled`

**Review by tomeresk — APPROVED** [review]

**Pull request merged**